[DOC-127] MVP for OSS Ray labels (#54254)
ray-project/ray: doc/source/ray-core/scheduling/index.rst

Scheduling
==========

This page provides an overview of how Ray decides to schedule tasks and actors to nodes.

.. DJS 19 Sept 2025: There should be an overview of all features and configs that impact scheduling here.
  This should include descriptions for default values and behaviors, and links to things like default labels or resource definitions that can be used for scheduling without customization.

Labels
------

Labels provide a simplified solution for controlling scheduling for tasks, actors, and placement group bundles using default and custom labels. See :doc:`./labels`.

> Labels are currently offered as an alpha feature. As this feature becomes stable, the Ray team recommends using labels instead of the following patterns:

Reviewer: ```suggestion
Labels are currently offered as an beta feature. As this feature becomes stable, the Ray team recommends using labels instead of the following patterns:
```

Scheduling
==========

This page provides an overview of how Ray decides to schedule tasks and actors to nodes.

.. DJS 19 Sept 2025: There should be an overview of all features and configs that impact scheduling here.
  This should include descriptions for default values and behaviors, and links to things like default labels or resource definitions that can be used for scheduling without customization.

Labels
------

Labels provide a simplified solution for controlling scheduling for tasks, actors, and placement group bundles using default and custom labels. See :doc:`./labels`.

> Labels are currently offered as an beta feature. As this feature becomes stable, the Ray team recommends using labels instead of the following patterns:

Reviewer 2: ```suggestion
Labels are a beta feature. As this feature becomes stable, the Ray team recommends using labels to replace the following patterns:
```

- NodeAffinitySchedulingStrategy when `soft=false`. Use the default `ray.io/node-id` label instead.
- The `accelerator_type` option for tasks and actors. Use the default `ray.io/accelerator-type` label instead.

.. note:: 

>   A legacy pattern recommended using custom resources for label-based scheduling. We now recommend only using custom resources when you need to manage scheduling using numeric values. 

Reviewer 2: ```suggestion
  A legacy pattern recommended using custom resources for label-based scheduling. Anyscale now recommends only using custom resources when you need to manage scheduling using numeric values. 
```

Author: This is Ray docs, so the recommender is "we" IIUC.
